src/test/java/uk/gov/hmcts/ccd/v2/external/controller/CaseControllerTestIT.java

```diff
@@ -798,4 +798,106 @@ private SupplementaryDataUpdateRequest createSupplementaryDataIncrementRequest()
+                .header(REQUEST_ID, REQUEST_ID_VALUE)
+                .contentType(JSON_CONTENT_TYPE))
+            .andReturn();
+
+        org.assertj.core.api.Assertions.assertThat(mvcResult.getResponse())
+            .isNotNull()
+            .satisfies(response -> org.assertj.core.api.Assertions.assertThat(response.getStatus()).isEqualTo(200));
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_cases.sql"})
+    void testShouldGetLinkedCasesNoCaseReferenceShouldReturn400() throws Exception {
+        final String caseId = "abc";
+        final String URL = "/getLinkedCases/" + caseId;
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+        final MvcResult mvcResult = mockMvc.perform(get(URL)
+                .header(REQUEST_ID, REQUEST_ID_VALUE)
+                .contentType(JSON_CONTENT_TYPE))
+            .andReturn();
+
+        org.assertj.core.api.Assertions.assertThat(mvcResult.getResponse())
+            .isNotNull()
+            .satisfies(response -> org.assertj.core.api.Assertions.assertThat(response.getStatus()).isEqualTo(400));
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_cases.sql"})
+    void testShouldGetLinkedCasesReturn404WhenCaseDoesNotExist() throws Exception {
+        final String caseId = "4259907353529155";
+        final String URL = "/getLinkedCases/" + caseId;
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+        final MvcResult mvcResult = mockMvc.perform(get(URL)
+                .header(REQUEST_ID, REQUEST_ID_VALUE)
+                .contentType(JSON_CONTENT_TYPE))
+            .andReturn();
+
+        org.assertj.core.api.Assertions.assertThat(mvcResult.getResponse())
+            .isNotNull()
+            .satisfies(response -> org.assertj.core.api.Assertions.assertThat(response.getStatus()).isEqualTo(404));
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_cases.sql"})
+    void testShouldGetLinkedCasesStartRecordNumberNotNumericReturn400() throws Exception {
+        final String caseId = "1504259907353529";
+        final String URL = "/getLinkedCases/" + caseId + "?startRecordNumber=A";
+        mapper.configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+        final MvcResult mvcResult = mockMvc.perform(get(URL)
+                .header(REQUEST_ID, REQUEST_ID_VALUE)
+                .contentType(JSON_CONTENT_TYPE))
+            .andReturn();
+
+        org.assertj.core.api.Assertions.assertThat(mvcResult.getResponse())
+            .isNotNull()
+            .satisfies(response -> org.assertj.core.api.Assertions.assertThat(response.getStatus()).isEqualTo(400));
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = {"classpath:sql/insert_cases.sql"})
```

Comment:
Dont think we need this here ?

Reply:
Pushed Change to RDM_13138 branch https://github.com/hmcts/ccd-data-store-api/commit/4cabb90d98b54e323f0495166a5bf3dc407e9eda